xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-ui/src/main/resources/Main/SolrSearchConfig.xml

```diff
@@ -51,6 +51,13 @@
     'OBJECT': 'objcontent',
     'OBJECT_PROPERTY': 'propertyvalue'
   },
+  'phraseFields': 'title_en^200.0 title__^200.0
```

Comment:
You should use just "title" which gets expanded automatically to "title_en", like for the query fields.

Reply:
I couldn't get it to expand, hence why I hard coded it. If I didn't explicitly specify as I've done then the terms wouldn't appear at all in the query string as displayed in debug mode.

Reply:
The code that is responsible for expanding the field name is [this](https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-server/xwiki-platform-search-solr-server-plugin/src/main/java/org/xwiki/search/solr/internal/XWikiDismaxQParserPlugin.java#L253). Indeed, it adds the alias only for the ``qf`` (query fields) parameter. Would be great if you could check if the ``pf`` parameter supports aliases (e.g. specified with ``f.%s.pf``).

Reply:
So... Done some research on the web and in the Solr documentation and aliasing doesn't seem to be possible for `pf` fields. Unsure what the best way around this is.

Reply:
OK, thanks for the investigation. In this case the solution I see is to write a small Velocity macro in ``SolrSearchConfig`` that gets as input parameters the field name (e.g. "title") and the weight (e.g. 200) and outputs ``title_ab^200 title_cd^200 ... title_yz^200 title__^200`` by reading the list of supported languages (locales) from the XWiki preferences. You can then call this macro like this:

    'phraseFields': "#expandField('title' 200) filename^45 #expandField('doccontent' 40) ..."

WDYT?

Reply:
Sounds good. I think I can manage that.

Reply:
Marius,
Can't figure out how to get the list of locales.
I've tried:
```
#set ($supportedLocales = $xwiki.getXWikiPreference('locales'))
#set ($supportedLocales = $services.localization.availableLocales)
#set ($supportedLocales = $xwiki.supportedLocales)
```
What am I missing?

Reply:
This should work:

    #set ($supportedLocales = $xwiki.getXWikiPreference('languages').split('\s*[|, ]\s*'))

The preference name is "languages" and not "locales" because it's an old preference that we didn't rename in order to not break backward compatibility.